tools/common-code/cmonitor_watcher.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/python3
+
+#
+# cmonitor_watcher.py
+#
+# Author: Satyabrata Bharati
+# Created: April 2022
+#
+import inotify.adapters
+import queue
+import os
+import time
+import logging
+from datetime import datetime
+
+exit_flag = False
+# =======================================================================================================
+# CgroupWatcher : Basic inotify class
+# =======================================================================================================
+class CgroupWatcher:
+    """
+    - Watch all files below a directory and notify an event for changes.
+    - Retrieves all the process and extract the process name "/proc/<pid>/stat.
+    - check the process name against the white-list given in the filter list.
+    - store the events in Queue.
+    """
+
+    def __init__(self, path, filter, timeout):
+        """Initialize CgroupWatcher
+        Args:
+          path: path to watch for events.
+          filter: white-list against which the process-event is filtered.
+
+        """
+        self.path = path
+        self.filter = filter
```

Comment:
from the documentation it's not clear what is the format of self.filter honestly. Is it a list of strings? please document